cg/meta/deliver.py

```diff
@@ -91,22 +93,27 @@ def deliver_files(self, case_obj: Family):
 
         link_obj: FamilySample
         for link_obj in link_objs:
-            sample_id: str = link_obj.sample.internal_id
-            sample_name: str = link_obj.sample.name
-            LOG.debug("Fetch last version for sample bundle %s", sample_id)
-            if self.delivery_type == "fastq":
-                last_version: hk_models.Version = self.hk_api.last_version(bundle=sample_id)
-            if not last_version:
-                if self.skip_missing_bundle:
-                    LOG.info(f"Could not find any version for {sample_id}")
-                    continue
-                raise SyntaxError(f"Could not find any version for {sample_id}")
-            self.deliver_sample_files(
-                case_id=case_id,
-                case_name=case_name,
-                sample_id=sample_id,
-                sample_name=sample_name,
-                version_obj=last_version,
+            if link_obj.sample.sequencing_qc or self.deliver_failed_samples:
+                sample_id: str = link_obj.sample.internal_id
+                sample_name: str = link_obj.sample.name
+                LOG.debug("Fetch last version for sample bundle %s", sample_id)
+                if self.delivery_type == "fastq":
+                    last_version: hk_models.Version = self.hk_api.last_version(bundle=sample_id)
+                if not last_version:
+                    if self.skip_missing_bundle:
+                        LOG.info(f"Could not find any version for {sample_id}")
+                        continue
+                    raise SyntaxError(f"Could not find any version for {sample_id}")
```

Comment:
When importing `HousekeeperVersionMissingError` here we get a circular import error. I spent about an hour looking into it and i suspect it has to do with the `cg/cli/generate/report/utils.py` module and it's tests. If it's okay with you I'll leave it as is, since I have not changed this line.

Reply:
👍